recipe/meta.yaml

```diff
@@ -51,6 +51,7 @@ outputs:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
         - {{ compiler('cuda') }}    # [cuda_compiler_version != "None"]
+        - cuda-libraries-dev  # [cuda_compiler_version != "None" and cuda_compiler_version.split(".")[0] >=12]
```

Comment:
Would use the specific libraries needed instead of the metapackage. The `-dev` libraries themselves have `run_exports` we would like to have to ensure they are correctly added to `requirements/run` for end users

Also we would want to list these in `requirements/host` (noted below). If we are using `requirements/build`, we can list in `build` for that case as well

As an example [`cupy` lists the following `-dev` libraries in `requirements/host`]( https://github.com/conda-forge/cupy-feedstock/blob/3e2f16dd92fc224cccd3f8875f673ce720148906/recipe/meta.yaml#L120-L132 ). Additionally [`cupy` lists `-dev` libraries in `requirements/build` for cross-compilation]( https://github.com/conda-forge/cupy-feedstock/blob/3e2f16dd92fc224cccd3f8875f673ce720148906/recipe/meta.yaml#L92-L104 ). Note that [`cupy` defines `cuda_major` initially]( https://github.com/conda-forge/cupy-feedstock/blob/3e2f16dd92fc224cccd3f8875f673ce720148906/recipe/meta.yaml#L10-L14 )

As a starting point, it may make sense to just copy over what `cupy` has in these cases and then tweak that to add/remove other libraries that come up. Am guessing there is a fair bit of overlap in what `torch` & `cupy` use here